pkg/lang/javascript/plugin_persist.go

```diff
@@ -161,27 +161,27 @@ func (p *persister) handleFile(f *core.SourceFile, unit *core.ExecutionUnit) ([]
 			errs.Append(core.NewCompilerError(f, annot, errors.New("'id' is required")))
 		}
 
-		var resource core.Construct
+		var construct core.Construct
 		var err, runtimeErr, transformErr error
 		switch keyType.(type) {
 		case *core.Kv:
-			resource, transformErr = p.transformKV(unit, f, annot, pResult)
+			construct, transformErr = p.transformKV(unit, f, annot, pResult)
 			runtimeErr = p.runtime.AddKvRuntimeFiles(unit)
 		case *core.Fs:
 			var envVarName string
-			resource, envVarName, transformErr = p.transformFS(unit, f, annot, pResult)
+			construct, envVarName, transformErr = p.transformFS(unit, f, annot, pResult)
 			runtimeErr = p.runtime.AddFsRuntimeFiles(unit, envVarName, cap.ID)
 		case *core.Secrets:
-			resource, transformErr = p.transformSecret(f, annot, pResult)
+			construct, transformErr = p.transformSecret(f, annot, pResult)
 			runtimeErr = p.runtime.AddSecretRuntimeFiles(unit)
 		case *core.Orm:
-			resource, transformErr = p.transformORM(unit, f, annot, pResult)
+			construct, transformErr = p.transformORM(unit, f, annot, pResult)
 			runtimeErr = p.runtime.AddOrmRuntimeFiles(unit)
 		case *core.RedisCluster:
-			resource, transformErr = p.transformRedis(unit, f, annot, pResult, keyType)
+			construct, transformErr = p.transformRedis(unit, f, annot, pResult, keyType)
 			runtimeErr = p.runtime.AddRedisClusterRuntimeFiles(unit)
```

Comment:
good renaming, just curious as to what made you look at this plugin?

Reply:
I was just backtracking to understand the the how we handle secret detection.